editor/src/clj/editor/keymap.clj

```diff
@@ -700,14 +701,20 @@
 (defn install!
   "Install a keymap on a scene, replacing any predefined accelerators"
   [keymap ^Scene scene execute-fn]
-  (doto (.getAccelerators scene)
-    (.clear)
-    (.putAll (persistent!
-               (reduce-kv
-                 (fn [acc shortcut commands]
-                   (assoc! acc shortcut #(execute-fn commands)))
-                 (transient {})
-                 (:shortcut->commands keymap)))))
+  (when-let [old-handler (.get (.getProperties scene) ::keymap)]
+    (.removeEventHandler scene KeyEvent/KEY_PRESSED old-handler))
+  (let [{:keys [shortcut->commands]} keymap
+        new-handler (reify EventHandler
+                      (handle [_ e]
+                        (reduce-kv
+                          (fn [_ ^KeyCombination shortcut commands]
+                            (when (.match shortcut e)
+                              (.consume e)
+                              (reduced (execute-fn commands))))
+                          nil
+                          shortcut->commands)))]
```

Comment:
Fun fact: this is semantically the same code that JavaFX uses internally when invoking a scene accelerator. Yes, accelerators are a map, but JavaFX also iterates over every entry in an attempt to `.match` a key combination with a handler:
https://github.com/openjdk/jfx/blob/3fdd21386d6db96294fcecd80afc25d09732c067/modules/javafx.graphics/src/main/java/com/sun/javafx/scene/KeyboardShortcutsHandler.java#L372-L387